modules/highgui/src/window_gtk.cpp

```diff
@@ -1973,7 +1982,25 @@ static gboolean icvOnMouse( GtkWidget *widget, GdkEvent *event, gpointer user_da
 //        if((unsigned)pt.x < (unsigned)(image_widget->original_image->width) &&
 //           (unsigned)pt.y < (unsigned)(image_widget->original_image->height) )
         {
-            int flags = (state & GDK_SHIFT_MASK ? CV_EVENT_FLAG_SHIFTKEY : 0) |
+            int flags = 0;
+            if (state & GDK_SCROLL_MASK) {
+                cv_event = CV_EVENT_MOUSEMOVE;
+                int orient = CV_EVENT_MOUSEWHEEL;
+                switch(event->scroll.direction) {
+                case GDK_SCROLL_RIGHT: orient = CV_EVENT_MOUSEHWHEEL;
+                case GDK_SCROLL_UP:    flags |= ((1 << 16) | orient);
+                    break;
+                case GDK_SCROLL_LEFT:  orient = CV_EVENT_MOUSEHWHEEL;
+                case GDK_SCROLL_DOWN:  flags |= ((-1 << 16) | orient);
+                    break;
+#if (GTK_MAJOR_VERSION > 3 or (GTK_MAJOR_VERSION == 3 and GTK_MINOR_VERSION >= 4))
```

Comment:
@alalek, sure.
Due mix of programming languages in my head and the night was born this staff :)
I'll fix it soon.